modules/test/playwright/tests/object-web/utils/asyncFilter.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator} from '@playwright/test';
+
+interface asyncFilter {
+	array: Locator[];
+	predicate: (value: Locator) => Promise<boolean>;
+}
+
+export async function asyncFilter({array, predicate}: asyncFilter) {
+	const results = await Promise.all(array.map(predicate));
+
+	return array.filter((_, index) => results[index]);
+}
```

Comment:
Hey @igor-franca, as we talked personally, I created this [commit](https://github.com/danvb112/liferay-portal/commit/5af0cd144a1cc942cf87e9a4c0463e2bc94f3cc8) by extracting the logic of the async Filter and async Find functions into a generic class, so we can reuse this code in other places as well :smile: